Include/httpClient/httpClient.h

```diff
@@ -679,101 +286,109 @@ HCHttpCallRequestGetRequestBodyString(
 HC_API void HC_CALLING_CONV
 HCHttpCallRequestSetHeader(
     _In_ HC_CALL_HANDLE call,
-    _In_ PCSTR_T headerName,
-    _In_ PCSTR_T headerValue
-    );
-
-/// <summary>
-/// Get a request header for the HTTP call for a given header name
-/// </summary>
-/// <param name="call">The handle of the HTTP call</param>
-/// <param name="headerName">request header name for the HTTP call</param>
-/// <param name="headerValue">request header value for the HTTP call</param>
-HC_API void HC_CALLING_CONV
-HCHttpCallRequestGetHeader(
-    _In_ HC_CALL_HANDLE call,
-    _In_ PCSTR_T headerName,
-    _Out_ PCSTR_T* headerValue
-    );
-
-/// <summary>
-/// Gets the number of request headers in the the HTTP call
-/// </summary>
-/// <param name="call">The handle of the HTTP call</param>
-/// <param name="numHeaders">the number of request headers in the the HTTP call</param>
-HC_API void HC_CALLING_CONV
-HCHttpCallRequestGetNumHeaders(
-    _In_ HC_CALL_HANDLE call,
-    _Out_ uint32_t* numHeaders
-    );
-
-/// <summary>
-/// Gets the request headers at specific zero based index in the the HTTP call.
-/// Use HCHttpCallRequestGetNumHeaders() to know how many request headers there are in the HTTP call.
-/// </summary>
-/// <param name="call">The handle of the HTTP call</param>
-/// <param name="headerIndex">Specific zero based index of the request header</param>
-/// <param name="headerName">Request header name for the HTTP call</param>
-/// <param name="headerValue">Request header value for the HTTP call</param>
-HC_API void HC_CALLING_CONV
-HCHttpCallRequestGetHeaderAtIndex(
-    _In_ HC_CALL_HANDLE call,
-    _In_ uint32_t headerIndex,
-    _Out_ PCSTR_T* headerName,
-    _Out_ PCSTR_T* headerValue
+    _In_ PCSTR headerName,
+    _In_ PCSTR headerValue
     );
 
 /// <summary>
 /// Sets if retry is allowed for this HTTP call
 /// Defaults to true 
 /// </summary>
-/// <param name="call">The handle of the HTTP call</param>
+/// <param name="call">The handle of the HTTP call.  Pass nullptr to set the default for future calls</param>
 /// <param name="retryAllowed">If retry is allowed for this HTTP call</param>
 HC_API void HC_CALLING_CONV
 HCHttpCallRequestSetRetryAllowed(
-    _In_ HC_CALL_HANDLE call,
+    _In_opt_ HC_CALL_HANDLE call,
     _In_ bool retryAllowed
```

Comment:
Bool is not a type in C

Reply:
Should be alright.  https://stackoverflow.com/a/1608349/7655781

Reply:
make sure you include stdbool.h then

I've definitely seen vs barf on it without any includes